src/main/java/com/kobylynskyi/graphql/codegen/GraphQLDocumentParser.java

```diff
@@ -52,6 +54,25 @@ static ExtendedDocument getDocument(MappingConfig mappingConfig, List<String> sc
         return extendedDocumentBuilder.build();
     }
 
+    static ExtendedDocument getDocumentFromIntrospectionResult(MappingConfig mappingConfig, String introspectionResult) throws IOException {
+        String introspectionResultContent = Utils.getFileContent(introspectionResult);
+        Map<String, Object> introspectionResultMap = Utils.OBJECT_MAPPER.readValue(introspectionResultContent,
+                new TypeReference<Map<String, Object>>() {
+                });
+        // unwrapping "data" (in case such GraphQL response supplied)
+        if (introspectionResultMap.containsKey("data") && !introspectionResultMap.containsKey("__schema")) {
```

Comment:
Do we need to check that the introspection result doesn't contain "__schema" in order to get the "data"?

I think it is useless, because the introspection query contains either "data" or "__schema".

It means that, if we have an introspection query like that (which is wrong):

```
{
  "data": ...,
  "__shema": ...
}
```

-> we will ignore the "data" object, because `new IntrospectionResultToSchema().createSchemaDefinition(introspectionResultMap)` will only take the "__schema" object.

I think wee need to do these steps, in this order:
1) If the map contains "data", unwrap as you did ;
2) then, check that the map contains the key "__schema" (otherwise throw exception?) ;
3) call `new IntrospectionResultToSchema().createSchemaDefinition(introspectionResultMap)`

Reply:
Thanks, @lmartellotto !
Just so you know, `new IntrospectionResultToSchema().createSchemaDefinition(introspectionResultMap)` performs the check for `"__schema"`.